tools/react-integration-tester/src/shared.ts

```diff
@@ -0,0 +1,255 @@
+    const loaded = require(configPath) as unknown;
+    const conf: Config = (loaded as { default?: Config })?.default ?? (loaded as Config);
+    const perVersion = conf?.react?.[String(reactVersion)] ?? {};
+    overrides = perVersion;
+  }
+
+  const merged = {
+    commands: { ...defaults.commands },
+    dependencies: { ...defaults.dependencies },
+  };
+  if (overrides?.commands) {
+    // Map camelCase keys from config to script names used in builtins
+    const map: Record<string, string> = {
+      test: 'test',
+      e2e: 'e2e',
+      typeCheck: 'type-check',
+    };
+    for (const [k, v] of Object.entries(overrides.commands)) {
+      const key = map[k] ?? k;
+      if (v) {
+        merged.commands[key] = v as string;
+      }
+    }
+  }
+  if (overrides?.dependencies) {
+    merged.dependencies = { ...merged.dependencies, ...overrides.dependencies };
+  }
+  return merged;
+}
+
+/**
+ * Prepare template by filtering commands based on the origin project's setups (Jest/Cypress) and
+ * reporting which setups exist. Keeps getMergedTemplate pure while providing a convenient wrapper.
+ */
+export function getPreparedTemplate(
+  reactVersion: ReactVersion,
+  configPath: string,
+  projectPaths: {
+    packageJson: string;
+    jestConfig: string | null;
+    cypressConfig: string | null;
+    tsConfig: string | null;
+  },
+): {
+  commands: Record<string, string>;
+  dependencies: Record<string, string>;
+  hasJestSetup: boolean;
+  hasCypressSetup: boolean;
+} {
+  const merged = getMergedTemplate(reactVersion, configPath);
+  const hasJestSetup = Boolean(projectPaths.jestConfig);
+  const hasCypressSetup = Boolean(projectPaths.cypressConfig);
+
+  const filteredCommands = Object.fromEntries(
+    Object.entries(merged.commands).filter(([commandKey]) => {
+      if (!hasCypressSetup && /^e2e($|:)/.test(commandKey)) {
+        return false;
+      }
+      if (!hasJestSetup && /^test($|:)/.test(commandKey)) {
+        return false;
+      }
+      return true;
+    }),
+  ) as Record<string, string>;
+
+  return {
+    commands: filteredCommands,
+    dependencies: merged.dependencies,
+    hasJestSetup,
+    hasCypressSetup,
+  } as const;
+}
+
+export function runCmd(command: string, opts: { cwd: string; env?: NodeJS.ProcessEnv }): Promise<void> {
+  return new Promise((resolvePromise, reject) => {
+    const localBin = join(opts.cwd, 'node_modules', '.bin');
+    const parentBin = join(opts.cwd, '..', 'node_modules', '.bin');
+    const pathKey = process.platform === 'win32' ? 'Path' : 'PATH';
+    const existingPath = process.env[pathKey] ?? '';
+    const mergedPath = [localBin, parentBin, existingPath]
+      .filter(Boolean)
+      .join(process.platform === 'win32' ? ';' : ':');
+    // Help Node resolve modules like '@swc/jest' from parent/react-root or repo root
+
+    const existingNodePath = (process.env as NodeJS.ProcessEnv & { NODE_PATH?: string }).NODE_PATH ?? '';
```

Comment:
this is on purpose to follow nodes the node_modules resolution algo, so we don't have to install all dependencies for integration test ( except those that are required to create isolated environment for the test )